c10/util/ArrayRef.h

```diff
@@ -22,10 +22,87 @@
 
 #include <array>
 #include <iterator>
+#include <memory>
+#include <type_traits>
 #include <vector>
 
 namespace c10 {
 
+
+
+struct TORCH_API SymbolicIntImpl {
+  bool virtual isSymbolicInt(int64_t data_) = 0;
+
+  // I can't overload + for primitive types
+  int64_t virtual add(int64_t s1, int64_t s2) = 0;
+  bool virtual equal(int64_t s1, int64_t s2) = 0;
```

Comment:
Taking an `int64_t` here seems wrong, you should take `SymbolicInt` (and rely on the insides to "unbox" the integers), instead of passing around the raw repr.